docs/getting-started.md

```diff
@@ -0,0 +1,117 @@
+uses internally). Namely also offers an open-source docker-based solution
+[namely/docker-protoc](https://github.com/namely/docker-protoc) with support
+for many languages.
+
+## Implement a "Write Side"
+
+Drawing from CQRS terminology, COS defines a [writeside interface](https://github.com/namely/chief-of-state-protos/blob/master/chief_of_state/v1/writeside.proto) that you must implement to define how state
+is created and updated.
+
+### Methods
+`HandleCommand` accepts a command (think "request") and a prior state and returns an event.
+For example, given a command to `UpdateUserEmail` and a User, this RPC might return
+an event called `UserEmailUpdated`. This method is allowed to throw exceptions
+(as gRPC error statuses) if the inbound command (request) is invalid.
+
+`HandleEvent` accepts an event and the prior state of an entity and returns a new state.
+For example, given a UserEmailUpdated event and a User, this RPC would return a
+new User instance with the email updated. It's advisable to do all your
+validations in `HandleCommand` such that this method is unlikely to encounter
+invalid data. If `HandleCommand` fails, `HandleEvent` is not called for that
+command.
+
+### Registration
+Register your write handler service as COS environment variables like so:
+
+```yaml
+# set the host and port for your write handler gRPC server
+COS_WRITE_SIDE_HOST: localhost
+COS_WRITE_SIDE_PORT: 50051
+```
+
+(If you are new to setting up COS, see the full [configuration](docs/configuration.md) guide.)
+
+### Sending requests to COS
+Once you have implemneted a write handler and registered it with COS, you are
+able to start sending requests to chief-of-state with the chief-of-state
+[gRPC client methods](https://github.com/namely/chief-of-state-protos/blob/master/chief_of_state/v1/service.proto).
+
+`ProcessCommand` accepts a command (request) and an entity ID, and processes
+that command by forwarding it to your write handler methods. Commands for a given
+entity are guaranteed to be processed serially, but commands for unrelated entities
+can be processed in parallel.
+
+`GetState` accepts an entity ID and returns the latest state of that entity. Use
+this like a key/value lookup.
+
+## Implement a "Read Side"
+
+COS defines a [readside interface](https://github.com/namely/chief-of-state-protos/blob/master/chief_of_state/v1/readside.proto)
+and allows you to implement many "readsides" (from CQRS "read side") to
+stream your events and state to external destinations. For example, you might
+implement a readside that writes to your applications private elasticsearch
+cluster for serving fancy search queries, and you could implement a second
+readside that publishes your event to kafka for consumption by other services.
+
+The COS readside processor guarantees that events are served in the order they were
+persisted to the journal per entity. Offsets are tracked for per readside, so
```

Comment:
I think we should link or add a few words explaining what a journal is, as part of  the cqrs world

Reply:
Yeah, I wrestled with how in depth this doc should be.

Reply:
I don't think we should explain what Journal is in the CQRS world. There are a lot of doc out there explaining. We can just elaborate that COS help build event sourced/cqrs driven application and maybe and I mean maybe point to the some docs.